### PR TITLE
feat: add native addon to `cflipsign`

### PR DESCRIPTION
**Pull request by steff456**
Part of https://github.com/stdlib-js/todo/issues/1454

## Description

> What is the purpose of this pull request?

This pull request:

-   Adds native addon with benchmark, tests, examples and readme update
-   Adds two new macros to the Node API math binary module

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   Part of https://github.com/stdlib-js/todo/issues/1454

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/math/base/napi/binary/README.md [hunk 1]
```diff
@@ -707,6 +707,76 @@ The macro expects the following arguments:
...
 When used, this macro should be used **instead of** `NAPI_MODULE`. The macro includes `NAPI_MODULE`, thus ensuring Node-API module registration.
 
+#### STDLIB_MATH_BASE_NAPI_MODULE_ZD_Z( fcn )
+
+Macro for registering a Node-API module exporting an interface invoking a binary function accepting a double-precision complex floating-point number and a double-precision floating-point number and returning a double-precision complex floating-point number.
+
+```c
+#include "stdlib/complex/float64.h"
+#include "stdlib/complex/reim.h"
+#include <stdint.h>
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/napi/binary/README.md]:
```suggestion
```

File: lib/node_modules/@stdlib/math/base/napi/binary/README.md [hunk 2]
```diff
@@ -707,6 +707,76 @@ The macro expects the following arguments:
...
+When used, this macro should be used **instead of** `NAPI_MODULE`. The macro includes `NAPI_MODULE`, thus ensuring Node-API module registration.
+
+#### STDLIB_MATH_BASE_NAPI_MODULE_CF_C( fcn )
+
+Macro for registering a Node-API module exporting an interface invoking a binary function accepting a single-precision complex floating-point number and a single-precision floating-point number and returning a single-precision complex floating-point number.
+
+```c
+#include "stdlib/complex/float32.h"
+#include "stdlib/complex/reimf.h"
+#include <stdint.h>
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/napi/binary/README.md]:
```suggestion
```

File: lib/node_modules/@stdlib/math/base/napi/binary/include/stdlib/math/base/napi/binary.h [hunk 3]
```diff
@@ -396,6 +396,102 @@
...
 
+/**
+* Macro for registering a Node-API module exporting an interface invoking a binary function accepting a double-precision complex floating-point number and a double-precision floating-point number and returning a double-precision complex floating-point number.
+*
+* @param fcn   binary function
+*
+* @example
+* #include "stdlib/complex/float64.h"
+* #include "stdlib/complex/reim.h"
+* #include <stdint.h>
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/napi/binary/include/stdlib/math/base/napi/binary.h]:
```suggestion
```

File: lib/node_modules/@stdlib/math/base/napi/binary/include/stdlib/math/base/napi/binary.h [hunk 4]
```diff
@@ -396,6 +396,102 @@
...
+			NULL,                                                              \
+			&fcn                                                               \
+		);                                                                     \
+		assert( status == napi_ok );                                           \
+		return fcn;                                                            \
+	};                                                                         \
+	NAPI_MODULE( NODE_GYP_MODULE_NAME, stdlib_math_base_napi_zd_z_init )
+
+/**
+* Macro for registering a Node-API module exporting an interface invoking a binary function accepting a single-precision complex floating-point number and a single-precision floating-point number integer and returning a single-precision complex floating-point number.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/napi/binary/include/stdlib/math/base/napi/binary.h]:
```suggestion
* Macro for registering a Node-API module exporting an interface invoking a binary function accepting a single-precision complex floating-point number and a single-precision floating-point number and returning a single-precision complex floating-point number.
```

File: lib/node_modules/@stdlib/math/base/napi/binary/include/stdlib/math/base/napi/binary.h [hunk 5]
```diff
@@ -396,6 +396,102 @@
...
+*
+* @param fcn   binary function
+*
+* @example
+* #include "stdlib/complex/float32.h"
+* #include "stdlib/complex/reimf.h"
+* #include <stdint.h>
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/napi/binary/include/stdlib/math/base/napi/binary.h]:
```suggestion
```

File: lib/node_modules/@stdlib/math/base/special/cflipsign/manifest.json [hunk 6]
```diff
@@ -24,17 +26,56 @@
...
+			"./include"
+		  ],
+		  "libraries": [],
+		  "libpath": [],
+		  "dependencies": [
+			"@stdlib/math/base/napi/binary",
+			"@stdlib/complex/float64",
+			"@stdlib/complex/reim",
+			"@stdlib/math/base/assert/is-negative-zero",
+			"@stdlib/math/base/special/flipsign"
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/cflipsign/manifest.json]: Why is this needed for the implementation? I only see this being used in the benchmarks.

File: lib/node_modules/@stdlib/math/base/special/cflipsign/manifest.json [hunk 7]
```diff
@@ -24,17 +26,56 @@
...
-			"libraries": [],
-			"libpath": [],
-			"dependencies": [
-        "@stdlib/math/base/assert/is-negative-zero"
-      ]
+		  "task": "build",
+		  "src": [
+			"./src/main.c"
+		  ],
+		  "include": [
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/cflipsign/manifest.json]: Mixed TABS and spaces. Ensure EditorConfig is properly setup.

File: lib/node_modules/@stdlib/math/base/special/cflipsign/manifest.json [hunk 8]
```diff
@@ -24,17 +26,56 @@
...
+			"./include"
+		  ],
+		  "libraries": [],
+		  "libpath": [],
+		  "dependencies": [
+			"@stdlib/complex/float64",
+			"@stdlib/complex/reim",
+			"@stdlib/math/base/assert/is-negative-zero",
+			"@stdlib/math/base/special/flipsign"
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/cflipsign/manifest.json]: I don't believe this package is used in the example code or in the implementation.

File: lib/node_modules/@stdlib/math/base/special/cflipsign/benchmark/c/benchmark.c [hunk 9]
```diff
@@ -105,7 +105,7 @@ double benchmark() {
 	for ( i = 0; i < ITERATIONS; i++ ) {
 		v = ( 1000.0*rand_double() ) - 500.0;
 		x = v + v*I;
-		y = stdlib_base_cflipsign( x, -v );
+		y = stdlib_base_flipsign( creal( x ), -v ) + stdlib_base_flipsign( cimag( x ), -v )*I;
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/cflipsign/benchmark/c/benchmark.c]: Instead of using `base_flipsign` here, I'd say we can just remove this benchmark file and its associated Makefile.

File: lib/node_modules/@stdlib/math/base/special/cflipsign/manifest.json (hunk 6 above)
Comment (kgryte) [reply lib/node_modules/@stdlib/math/base/special/cflipsign/manifest.json]: Per my above comment, we can prob remove the dep on `flipsign` altogether by removing the vanilla C benchmark.

**Review by kgryte — CHANGES_REQUESTED** [review]
@steff456 Thanks for working on this. Left a couple of comments. One to remove the vanilla C benchmark. And the other to fix indentation in the `manifest.json` file. Once resolved, this PR should be ready for merge.

File: lib/node_modules/@stdlib/math/base/special/cflipsign/manifest.json (hunk 7 above)
Comment (kgryte) [reply lib/node_modules/@stdlib/math/base/special/cflipsign/manifest.json]: Update: mixed tabs and spaces seems to stem from the `manifest.json` file using old conventions and then the more recent EditorConfig conventions overriding, leading to mixed behavior. We should do a sweep across the project to normalize indentation in these files to prevent further copy-paste errors.

In the meantime, if you can convert indentation to spaces, that would be appreciated!

Comment (steff456) [reply lib/node_modules/@stdlib/math/base/special/cflipsign/manifest.json]: I'll open an issue for this one! 

Update: I created the issue here, https://github.com/stdlib-js/todo/issues/1648

File: lib/node_modules/@stdlib/math/base/special/cflipsign/examples/c/example.c [hunk 10]
```diff
@@ -17,19 +17,32 @@
...
+#include "stdlib/complex/float64.h"
+#include "stdlib/complex/reim.h"
 #include <stdio.h>
-#include <complex.h>
+
 
 int main( void ) {
-	// cppcheck-suppress nanInArithmeticExpression
-	double complex x[] = { 3.14+1.5*I, -3.14-1.5*I, 0.0+0.0*I, 0.0/0.0+0.0/0.0*I };
+  // cppcheck-suppress nanInArithmeticExpression
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/cflipsign/examples/c/example.c]:
```suggestion
```

**Review by kgryte — APPROVED** [review]
